test/redis-invalidation.test.js

```diff
@@ -0,0 +1,299 @@
+'use strict'
+
+const { test, teardown, beforeEach } = require('tap')
+const fastify = require('fastify')
+const mercurius = require('mercurius')
+const cache = require('..')
+const Redis = require('ioredis')
+const { request } = require('./helper')
+
+const redisClient = new Redis()
+
+teardown(async () => {
+  await redisClient.quit()
+})
+
+beforeEach(async () => {
+  await redisClient.flushall()
+})
+
+test('redis invalidation', async () => {
+  const schema = `
+    type Query {
+      get (id: Int): String
+      search (id: Int): String
+    }
+    type Mutation {
+      set (id: Int): String
+    }
+  `
+  const resolvers = {
+    Query: {
+      async get (_, { id }) {
+        return 'get ' + id
+      },
+      async search (_, { id }) {
+        return 'search ' + id
+      }
+    },
+    Mutation: {
+      async set (_, { id }) {
+        return 'set ' + id
+      }
+    }
+  }
+
+  test('should remove storage keys by references', async t => {
+    t.plan(7)
```

Comment:
in async tests, unless you're doing assertions in callbacks (which you're not doing in this test but you are in others) t.plan is redundant and unnecessary. the test will complete when the returned promise settles. iow, you can safely remove it from this test and any other test where you're not asserting in callbacks

Reply:
Nice! I didn't know it. I am going to remove the unused ones.

Reply:
Fixed.